tests/phpunit/base-schema.php

```diff
@@ -0,0 +1,100 @@
+<?php
+namespace Elementor\Testing;
+
+use Elementor\Core\Utils\Collection;
+use Elementor\Tracker;
+use JsonSchema\Exception\ValidationException;
+use JsonSchema\SchemaStorage;
+use JsonSchema\Validator;
+
+class Base_Schema extends Elementor_Test_Base {
+	const HTTP_USER_AGENT = 'test-agent';
+
+	public function setUp() {
+		parent::setUp();
+
+		// Required by `Tracker::get_tracking_data`.
+		$_SERVER['HTTP_USER_AGENT'] = self::HTTP_USER_AGENT;
+	}
+
+	protected function validate_against_schema( $data ) {
+		// Since the usage system represents objects as array instead of stdClass.
+		$data = json_decode( json_encode( $data ) );
+
+		// Validate
+		$validator = new Validator;
+		$validator->validate( $data, (object) [ '$ref' => 'file://' . ELEMENTOR_PATH . 'schemas/usage.json' ] );
```

Comment:
maybe the `ref` path should be dynamic to let other plugins use it.

Reply:
Handled